### PR TITLE
Added separate Callbacks for the forward and adjoint solves

### PR DESCRIPTION
**Pull request by PiotrSokol**
As discussed on Slack, I thought a separate callback for the forward and adjoint might be nifty for things like tracking gradients.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Yeah this is a cool idea! Needs docs though.

**Comment by PiotrSokol** [comment]
No problem -- I'll get to it probably only early November though.

**Comment by PiotrSokol** [comment]
@ChrisRackauckas ok, added very brief to neural_ode

File: src/Flux/neural_de.jl [hunk 1]
```diff
@@ -1,26 +1,50 @@
...
-                      args...;kwargs...)
-  dudt_(u,p,t) = model(u)
-  g(u,p,t) = mp.*u
-  prob = SDEProblem(dudt_,g,param(x),tspan,nothing)
-  # TODO could probably use vcat rather than collect here
-  solve(prob, args...; kwargs...) |> Tracker.collect
-end
+function neural_ode(model,x,tspan,
+                    args...;kwargs...)
+  """
```
Comment (ChrisRackauckas) [inline src/Flux/neural_de.jl]: docstrings go before the method definition

File: src/Flux/neural_de.jl [hunk 2]
```diff
@@ -1,26 +1,50 @@
...
+  Ref
+  [1]L. S. Pontryagin, Mathematical Theory of Optimal Processes. CRC Press, 1987.
+
+  Arguments
+  ≡≡≡≡≡≡≡≡
+  model::Chain defines the ̇x
+  x<:AbstractArray initial value x(t₀)
+  args arguments passed to ODESolve
+  kwargs key word arguments passed to ODESolve; accepts an additional key
+      :callback_adj in addition to :callback. The Callback :callback_adj
```
Comment (ChrisRackauckas) [inline src/Flux/neural_de.jl]: this should be mentioned in the README which is our current documentation

**Comment by ChrisRackauckas** [comment]
Tests fail

**Comment by PiotrSokol** [comment]
@ChrisRackauckas , it seems the error in Travis CI and appveyor stem from the fact that the `Project.toml` doesn't specify `Plots`. Having gone through the error messages, that seems to be the only thing crashing. The adjoint 'looks' good.

**Comment by ChrisRackauckas** [comment]
Rebase to master. That has already been fixed.

File: README.md [hunk 3]
```diff
@@ -632,12 +632,12 @@ The major options to keep in mind are:
   which are dominating the runtime, then the backpropogation can be made trivial by using mutation, and thus `f(u,p,t)` is
   more efficient. One example which falls into this case is the neural ODE which has large matrix multiply operations. However,
   if the neural network is a small portion of the equation and there is heavy reliance on directly specified nonlinear forms
-  in the differential equation, `f(du,u,p,t)` with the option `sense=SensitivityAlg(autojacvec=false)` may be preferred. 
+  in the differential equation, `f(du,u,p,t)` with the option `sense=SensitivityAlg(autojacvec=false)` may be preferred.
 - `diffeq_adjoint` currently only applies to ODEs, though continued development will handle other equations in the future.
 - `diffeq_adjoint` has O(1) memory with the default `backsolve`. However, it is known that this is unstable on many equations
-  with high enough stiffness (this is a fundamental fact of the numerics, see
```
Comment (ChrisRackauckas) [inline README.md]: try to keep these kinds of things in a separate PR. Gets in the way.

**Comment by ChrisRackauckas** [comment]
Thanks!
